lib/internal/modules/esm/loader.js

```diff
@@ -4,150 +4,195 @@
 const ModuleMap = require('internal/modules/esm/module_map');
 const ModuleJob = require('internal/modules/esm/module_job');
 
 const {
   defaultResolve,
   DEFAULT_CONDITIONS,
 } = require('internal/modules/esm/resolve');
-const { defaultGetFormat } = require('internal/modules/esm/get_format');
-const { defaultGetSource } = require(
-  'internal/modules/esm/get_source');
-const { defaultTransformSource } = require(
-  'internal/modules/esm/transform_source');
+const { defaultLoad } = require('internal/modules/esm/load');
 const { translators } = require(
   'internal/modules/esm/translators');
 const { getOptionValue } = require('internal/options');
 
-/* A Loader instance is used as the main entry point for loading ES modules.
+/**
+ * An ESMLoader instance is used as the main entry point for loading ES modules.
  * Currently, this is a singleton -- there is only one used for loading
- * the main module and everything in its dependency graph. */
-class Loader {
-  constructor() {
-    // Methods which translate input code or other information
-    // into es modules
-    this.translators = translators;
-
-    // Registry of loaded modules, akin to `require.cache`
-    this.moduleMap = new ModuleMap();
-
-    // Map of already-loaded CJS modules to use
-    this.cjsCache = new SafeWeakMap();
-
-    // This hook is called before the first root module is imported. It's a
-    // function that returns a piece of code that runs as a sloppy-mode script.
-    // The script may evaluate to a function that can be called with a
-    // `getBuiltin` helper that can be used to retrieve builtins.
-    // If the hook returns `null` instead of a source string, it opts out of
-    // running any preload code.
-    // The preload code runs as soon as the hook module has finished evaluating.
-    this._getGlobalPreloadCode = null;
-    // The resolver has the signature
-    //   (specifier : string, parentURL : string, defaultResolve)
-    //       -> Promise<{ url : string }>
-    // where defaultResolve is ModuleRequest.resolve (having the same
-    // signature itself).
-    this._resolve = defaultResolve;
-    // This hook is called after the module is resolved but before a translator
-    // is chosen to load it; the format returned by this function is the name
-    // of a translator.
-    this._getFormat = defaultGetFormat;
-    // This hook is called just before the source code of an ES module file
-    // is loaded.
-    this._getSource = defaultGetSource;
-    // This hook is called just after the source code of an ES module file
-    // is loaded, but before anything is done with the string.
-    this._transformSource = defaultTransformSource;
-    // The index for assigning unique URLs to anonymous module evaluation
-    this.evalIndex = 0;
-  }
+ * the main module and everything in its dependency graph.
+ */
+class ESMLoader {
+  /**
+   * Phase 0 of 2 in ESM loading. These are called once before any modules are
```

Comment:
```suggestion
   * Phase 1 of 3 in ESM loading. These are called once before any modules are
```

“Phase 0 of 2“ makes me wonder, are there only two phases or are there three?

Reply:
I chose `0 of 2` because preload is "before the first" and itself is not really a step—it doesn't do anything directly toward loading, only a metastep that influences the 2 main steps.

I don't have a strong opinion though.

Reply:
I think “0 of 2” is just very ambiguous, and doesn’t feel like correct English. If preloading isn’t a step, you could say “Precursor to ESM loading” here. Then later you could say “first part of ESM loading” and “second part of ESM loading” or something like that.